src/libopenrave/kinbody.cpp

```diff
@@ -5331,10 +5389,50 @@ bool CompareNonAdjacentFarthest(int pair0, int pair1)
 }
 
 const std::vector<int>& KinBody::GetNonAdjacentLinks(int adjacentoptions) const
+{
+    CHECK_INTERNAL_COMPUTATION;
+    if( _nNonAdjacentLinkCache & 0x80000000 ) {
+        std::vector<bool> adjacentLinkFlags;
+        _CalculateAdjacentLinkFlagsFromNonSelfCollidingPositionConfigurations(adjacentLinkFlags);
+
+        // Constructs _vNonAdjacentLinks[0]
+        _vNonAdjacentLinks[0].clear();
+        for( size_t ind0 = 0; ind0 < _veclinks.size(); ++ind0 ) {
+            for( size_t ind1 = ind0 + 1; ind1 < _veclinks.size(); ++ind1 ) {
+                if( !adjacentLinkFlags.at(_GetIndex1d(ind0, ind1)) && !AreAdjacentLinks(ind0, ind1) ) {
```

Comment:
The second check happens inside `_CalculateAdjacentLinkFlagsFromNonSelfCollidingPositionConfigurations` already.

Reply:
It is needed in a case no non-self-colliding position is given. The one in the calculation function is only for optimisation.

Reply:
True.